### ISSUE TITLE
call.on('close') function not firing on connected peer

### ISSUE THREAD

**Issue by maximus1127** [peers/peerjs#752]
I tried using version 1.3 as someone else reported this trouble on a separate thread. So I downgraded to 1.2 and the error is still happening. I'm setting up one-directional video calls. Once the call is initiated by the host, the streams starts, no issue. When the initiator fires the "call.close()" method on its browser, the stream stops. But on the receiver, it's listening for "call.on("close", callback) but it never fires. The only way to close it is to assign the call to a global variable and go to the console and type "globalVariable.close()". How can I resolve this? I've seen multiple other threads address this issue and there were no answers in those either, hence why i'm opening a new one. Just for reference, after about 5 seconds, the receiver has a log of "iceConnectionState changed to disconnected on the connection with myServerUUID". Any advice is appreciated.

**Comment by qikevinjkw** [comment]
I'm getting the same issue. When A calls B, and B closes their browser tab, A does not receive a 'close' event.

**Comment by morgan3d** [comment]
Note that not all browsers support close callbacks, either. (I think Firefox doesn't, for example). I use explicit pings and timeouts in my code rather than relying on the close event.

**Comment by maximus1127** [comment]
Yes I use this on chrome and opera with no success. I've validated this with other users within the peerjs telegram group and they have reported the same issue. I feel this is a legitimate bug with peerjs library

**Comment by morgan3d** [comment]
I think it is a WebRTC problem in the browser, and not a PeerJS problem. Peer does have its own ping system as a workaround, but it appears to be inactive in the current build.

**Comment by el07694** [comment]
+1 I tried with Chrome and with edge. With edge doesn't work

**Comment by maximus1127** [comment]
what about with chrome? that's what I'm trying it with

**Comment by el07694** [comment]
When a peer close it's tab, the video frozen.
We want close event to remove video element.

Doesn't work in Chrome either.

**Comment by mohinishsharma** [comment]
You could use window "beforeunload" event to terminal when user closes tab.

**Comment by MasterShi** [comment]
> When a peer close it's tab, the video frozen.
> We want close event to remove video element.
> 
> Doesn't work in Chrome either.

Same issue , doesn't fire `mediaConnection.on('close', callback)` in Chrome.

**Comment by newelement** [comment]
This also happens during navigator.mediaDevices.getDisplayMedia, in such when a user clicks the browser's native close screen share button. I guess it would be the same as if someone just closes the browser.

**Comment by naonvl** [comment]
any workaround on this?

**Comment by adityabhura** [comment]
I tried version 1.2.0 and hosted my app on heroku. It worked for me on all the browsers, but function didn't work on chrome while i was working locally but it worked in firefox.
